repose-aggregator/components/filters/keystone-v2-filter/src/main/scala/org/openrepose/filters/keystonev2/KeystoneRequestHandler.scala

```diff
@@ -186,7 +197,8 @@ class KeystoneRequestHandler(identityServiceUri: String, akkaServiceClient: Akka
       s"$identityServiceUri${GROUPS_ENDPOINT(forToken)}",
       (Map(CommonHttpHeader.AUTH_TOKEN.toString -> authenticatingToken,
         CommonHttpHeader.ACCEPT.toString -> MediaType.APPLICATION_JSON)
-        ++ traceId.map(CommonHttpHeader.TRACE_GUID.toString -> _)).asJava))
+        ++ traceId.map(CommonHttpHeader.TRACE_GUID.toString -> _)).asJava,
+      checkCache))
```

Comment:
Oh, I see.

Right, so the `KeystoneV2Filter` is responsible for retrying (because it can access the datastore, while the handler cannot, and has the logic for self-validating tokens). The `KeystoneRequestHandler` just makes the requests and returns the responses. The distance between the component which triggers this and the actual passing of the argument is only one method. So it is one more layer than would be preferable, but eh. It would require a fair amount of refactoring to drop this parameter and handle the retry in the `KeystoneRequestHandler`.

With the ability to use default values for method calls, I thought this actually came out fairly clean.